src/all_cats_photo.js

```diff
@@ -0,0 +1,74 @@
+(function () {
+    const selectElement = document.getElementById('friends');
+    const clickBtn = document.getElementById('clickBtn');
+    const popup = document.getElementById('popup');
+    const imagesContainer = document.getElementById('imagesContainer');
+
+    // Function to make API requests
+    async function getData(url) {
+        try {
+            const response = await fetch(url);
+            if (!response.ok) throw new Error(`Response status: ${response.status}`);
+            return await response.json();
+        } catch (error) {
+            console.error(error.message);
+            return error;
+        }
+    }
+
+    // Function to populate the dropdown with cat breeds
+    const getBreedArray = async () => {
+        const url = 'https://api.thecatapi.com/v1/breeds';
+        try {
+            const breedArray = await getData(url);
+            breedArray.forEach(breed => {
+                const option = document.createElement('option');
+                option.value = breed.id;
+                option.innerText = breed.name;
+                selectElement.appendChild(option);
```

Comment:
For performance considerations, it's better to change real DOM as less as possible. In this case we can append children to the `selectElement` only once, after all the options are collected.